src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -75,5 +62,75 @@ public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstruct
 
             return result;
         }
+
+        private ImmutableArray<BoundExpression> AccessTupleFields(BoundDeconstructionAssignmentOperator node, BoundExpression loweredRight, ArrayBuilder<LocalSymbol> temps, ArrayBuilder<BoundExpression> stores)
+        {
+            var tupleType = loweredRight.Type.IsTupleType ? loweredRight.Type : TupleTypeSymbol.Create((NamedTypeSymbol)loweredRight.Type);
+            var tupleElementTypes = tupleType.TupleElementTypes;
+
+            var numElements = tupleElementTypes.Length;
+            Debug.Assert(numElements == node.LeftVariables.Length);
+
+            CSharpSyntaxNode syntax = node.Syntax;
+
+            // save the loweredRight as we need to access it multiple times 
+            BoundAssignmentOperator assignmentToTemp;
+            var savedTuple = _factory.StoreToTemp(loweredRight, out assignmentToTemp);
```

Comment:
This is in general not a trivial optimization. RHS could be captured in a closure and changed between reads, could be used on the left, could be aliased via a byref local.

Simple redundant temp cases should be already handled in the codegen.